packages/kbn-esql-validation-autocomplete/src/definitions/aggs.ts

```diff
@@ -224,4 +226,71 @@ export const statsAggregationFunctionDefinitions: FunctionDefinition[] = [
         'from index | stats all_sorted_agents=mv_sort(values(agents.keyword))',
       ],
     },
+    {
+      name: 'top_list',
+      type: 'agg',
+      description: i18n.translate('kbn-esql-validation-autocomplete.esql.definitions.topListDoc', {
+        defaultMessage: 'Collects top N values per bucket.',
+      }),
+      supportedCommands: ['stats', 'metrics'],
+      signatures: [
+        {
+          params: [
+            {
+              name: 'field',
+              type: 'any',
+              noNestingFunctions: true,
+              optional: false,
+            },
+            {
+              name: 'limit',
+              type: 'number',
+              noNestingFunctions: true,
+              optional: false,
+              constantOnly: true,
+            },
+            {
+              name: 'order',
+              type: 'string',
+              noNestingFunctions: true,
+              optional: false,
+              constantOnly: true,
+              literalOptions: ['asc', 'desc'],
+            },
+          ],
+          returnType: 'any',
+        },
+      ],
+      examples: [
+        `from employees | stats top_salaries = top_list(salary, 10, "desc")`,
+        `from employees | stats date = top_list(hire_date, 2, "asc"), double = top_list(salary_change, 2, "asc"),`,
+      ],
+      validate: (fn: ESQLFunction) => {
```

Comment:
This custom validation should already be covered by the `literalOptions` stuff. Can we remove it?

Reply:
Currently, something does not work out, if I remove this validation it allows any string:

<img width="780" alt="image" src="https://github.com/elastic/kibana/assets/82822460/e8c81619-43be-4722-81d5-3195d4501b39">

Reply:
I guess the built-in validation somehow misses this. Could it be because this is an agg function?

Reply:
I've removed this custom logic to unblock, we can fix the built-in validation in a follow up.

Reply:
It's working as intended @vadimkibana 
<img width="736" alt="Screenshot 2024-07-01 at 4 00 18 PM" src="https://github.com/elastic/kibana/assets/315764/cefd8ae6-7a21-4366-9f80-6099c7e82a72">

It's a warning for now, but once Elasticsearch tightens up their validation for `mv_sort` (issue) we can change it to an error.

Reply:
Created an issue for our side: https://github.com/elastic/kibana/issues/187311

Reply:
Ah, I see, the test was asserting for error, not warning.